File: Competetive-Programming/TCS/Recursion_Start_to_Goal/start_to_goal.js
"use strict";

function isPossible(a, b, c, d) {
  console.log("execute", a, b, c, d);
  if (a == c && b == d) {
    return "Yes";
  } else if (a > c || b > d) {
    return "No";
  } else {
    isPossible(a + b, a + b, c, d);
    // isPossible(a, a + b, c, d);
  }
}

let result = isPossible(1, 1, 2, 2);
console.log(result);

File: Competetive-Programming/TCS/Recursion_Start_to_Goal/start_to_goal.cpp
// Online C++ compiler to run C++ program online
#include <bits/stdc++.h>

using namespace std;

string isPossible(int a, int b, int c, int d) {
    if(a==c && b==d){
        string result = "Yes";
        return result;
    }else if(a>c || b>d){
        string result = "No";
        return result;
    }else{
        isPossible(a+b,b,c,d);
        isPossible(a, a + b, c, d);
    }
}

int main() {
    // Write C++ code here
    int a = 1;
    int b = 1;
    int c = 2;
    int d = 2;
 string result = isPossible(a, b, c, d);    
    std::cout << result;

    return 0;
}